src/Microsoft.TestPlatform.CoreUtilities/Utilities/TimeSpanParser.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.VisualStudio.TestPlatform.Utilities
+{
+    public static class TimeSpanParser
+    {
+        static readonly Regex pattern = new Regex(@"(?<value>^\d+(?:\.\d+)?)\s*(?<suffix>ms|mil|m|h|d|s?[a-z]*)$", RegexOptions.IgnoreCase);
+
+        public static TimeSpan Parse(string time)
+        {
+            return TryParse(time, out var result) ? result : throw GetFormatException(time);
+        }
+
+        public static bool TryParse(string time, out TimeSpan result)
+        {
+            if (string.IsNullOrWhiteSpace(time))
+            {
+                result = TimeSpan.Zero;
+                return true;
+            }
+
+            var match = pattern.Match(time);
+            if (!match.Success)
+            {
+                result = TimeSpan.Zero;
+                return false;
+            }
+
+            var value = match.Groups["value"].Value;
+            if (!double.TryParse(value, out var number))
+            {
+                throw GetFormatException(value);
+            }
+
+            var suffix = match.Groups["suffix"].Value;
+            var c = StringComparison.OrdinalIgnoreCase;
+
+            // mil to distinguish milliseconds from minutes
+            // ""  when there is just the raw milliseconds value
+            if (suffix.StartsWith("ms", c) || suffix.StartsWith("mil", c) || suffix == string.Empty)
```

Comment:
Is it not  better to check for string.Empty first?

Reply:
Does not really matter here, this is not hot path, so the order has no real impact here. Won't fix.